lib/rspec/support/object_inspector.rb

```diff
@@ -3,23 +3,51 @@ module Support
     # Provide additional output details beyond what `inspect` provides when
     # printing Time, DateTime, or BigDecimal
     module ObjectInspector
+      DEFAULT_OPTIONS = {
+        :nested => false,
+        :joiner => ',',
+        :braces => true,
+        :trailing_comma => false
+      }
       # @api private
-      def self.inspect(object)
-        if Time === object
+      # rubocop:disable CyclomaticComplexity
+      # rubocop:disable MethodLength
+      def self.inspect(object, options=DEFAULT_OPTIONS)
+        case object
+        when Time
           format_time(object)
-        elsif defined?(DateTime) && DateTime === object
-          format_date_time(object)
-        elsif defined?(BigDecimal) && BigDecimal === object
-          "#{object.to_s 'F'} (#{object.inspect})"
-        elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
-          object.description
+        when Hash
+          format_hash(object, options)
```

Comment:
> The only issue I see here is that a Hash inside of an Array, will have both .inspect called on the hash, and then again once it's in the array, which I think will have weird formatting issues.

I'm open to suggestions on how to solve this without passing some information to one of the classes that lets it know that its being invoked inside of itself.